test/vagrant-service-manager/installer_test.rb

```diff
@@ -0,0 +1,89 @@
+require 'test_helper'
+require 'vagrant/util/downloader'
+
+module VagrantPlugins
+  module ServiceManager
+    describe Installer do
+      before do
+        @type = :docker
+        @machine = fake_machine
+        @env = @machine.env
+        @options = { box_version: 'adb', '--cli-version' => '1.11.0' }
+
+        # set bin and tmp directories
+        @base_test_path = "#{@env.data_dir}/service-manager/test"
+        ServiceManager.set_bin_dir("#{@base_test_path}/bin")
+        ServiceManager.set_temp_dir("#{@base_test_path}/tmp")
+        @bin_dir = ServiceManager.bin_dir
+        @temp_dir = ServiceManager.temp_dir
+
+        # Installer instance
+        @installer = Installer.new(@type, @machine, @env, @options)
+      end
+
+      after do
+        FileUtils.rmtree(@bin_dir) if File.directory? @bin_dir
+        FileUtils.rmtree(@temp_dir) if File.directory? @temp_dir
+      end
+
+      it 'should set default values properly' do
+        @installer.instance_variable_get('@binary_exists').must_equal true
+        @installer.instance_variable_get('@url').must_equal ''
+        @installer.instance_variable_get('@archive_file_path').must_equal ''
+        @installer.instance_variable_get('@type').must_equal @type
+        @installer.instance_variable_get('@machine').must_equal @machine
+        @installer.instance_variable_get('@ui').must_equal @env.ui
+        @installer.instance_variable_get('@temp_bin_dir').must_equal "#{@temp_dir}/#{@type.to_s}"
+        @installer.instance_variable_get('@binary_name').must_equal @type.to_s
+      end
+
+      it 'should update binary path on changing' do
+        expected_path = '/foo/bar/bin/docker'
+        options = @options.merge({ '--path' => expected_path })
+
+        @installer = Installer.new(@type, @machine, @env, options)
+        @installer.instance_variable_get('@path').must_equal expected_path
+      end
+
+      it 'should set docker binary handler' do
+        binary_object = @installer.instance_variable_get('@binary_object')
+        binary_object.must_be_instance_of DockerBinaryHandler
+      end
+
+      it 'should verify download url' do
+        expected_url = 'https://get.docker.com/builds/Linux/x86_64/docker-1.11.0.tgz'
```

Comment:
This is a Linux specific assertion. Test will not pass on OS X:

    Finished in 0.430219s, 16.2708 runs/s, 34.8660 assertions/s.
    
      1) Failure:
    VagrantPlugins::ServiceManager::Installer#test_0004_should verify download url [/Users/hardy/work/developer-platform/vagrant/vagrant-service-manager/test/vagrant-service-manager/installer_test.rb:57]:
    --- expected
    +++ actual
    @@ -1 +1 @@
    -"https://get.docker.com/builds/Linux/x86_64/docker-1.11.0.tgz"
    +"https://get.docker.com/builds/Darwin/x86_64/docker-1.11.0.tgz"
    
    
    
      2) Failure:
    VagrantPlugins::ServiceManager::Installer#test_0007_should also download docker less than 1.11.0 [/Users/hardy/work/developer-platform/vagrant/vagrant-service-manager/test/vagrant-service-manager/installer_test.rb:81]:
    --- expected
    +++ actual
    @@ -1 +1 @@
    -"https://get.docker.com/builds/Linux/x86_64/docker-1.9.1"
    +"https://get.docker.com/builds/Darwin/x86_64/docker-1.9.1"
    
    
    7 runs, 15 assertions, 2 failures, 0 errors, 0 skips